integration_test/tds/explain_test.exs

```diff
@@ -0,0 +1,12 @@
+defmodule Ecto.Integration.ExplainTest do
+  use Ecto.Integration.Case, async: true
+
+  alias Ecto.Integration.TestRepo
+  alias Ecto.Integration.Post
+
+  test "explain options" do
+    assert_raise(Tds.Error, "not supported yet", fn ->
```

Comment:
Looks like Tds doesn't support multiple result sets yet, which AFAIK is necessary to return results for explain statements on sqlserver, but I'm not so familiar with sqlserver and @mjaric could some light here.

Reply:
It is there already but not exposed to user code since %Tds.Result only provides single result. I think the best short term solution is to amend Tds.Result and add `statistics` attribute to it that should hold "execution plan". Then we can add flag in `query/4` as part of the option argument e.g. `explain: :estimate|:profile` so it know that it should attach last result in resultset to `statistics` attribute AND of course, before sending query it should wrap it with:

For `explain: :estimate`
```sql
SET STATISTICS XML ON 

/* original query goes here */

SET STATISTICS XML OFF 
```

It is worth mentioning that this will return only ESTIMATED query plan WITHOUT result, to get ACTUAL query plan + result it must be wrapped like this:

For `explain: :profile`
```sql

SET STATISTICS XML ON
SET STATISTICS PROFILE OON

/* original query here */

SET STATISTICS XML OFF 
SET STATISTICS PROFILE OFF
```

Reply:
@mjaric thanks for the feedback! I believe it's fine to raise for now and implement it as soon as https://github.com/livehelpnow/tds/issues/108 is defined and implemented.

`XML` is the only option available? Is there a way to get an ACTUAL query plan in a text format?

Reply:
There are 2 types of statistics that TDS driver already can log into logger if logger level is set to debug. Those are `SET STATISTICS IO ON` and `SET STATISTICS TIME ON`. Here is example

```elixir
SQL.query!(
        Repo,
        """
        SET STATISTICS IO ON
        SET STATISTICS TIME ON
        select top 1 * from [dbo].[test_table] where [description] like @1
        SET STATISTICS TIME OFF
        SET STATISTICS IO OFF
        """,
        ["3000"]
      ),
```

in log
```
06:46:36.250 [debug] (Tds.Info) Line 3 (Class 0) Table 'test_table'. Scan count 1, logical reads 2557, physical reads 0, read-ahead reads 0, lob logical reads 0, lob physical reads 0, lob read-ahead reads 0.
 
06:46:36.250 [debug] (Tds.Info) Line 3 (Class 0) 
 SQL Server Execution Times:
   CPU time = 16 ms,  elapsed time = 14 ms.
 
06:46:36.253 [debug] QUERY OK db=16.1ms decode=0.6ms queue=28.2ms idle=0.0ms
SET STATISTICS IO ON
SET STATISTICS TIME ON
select top 1 * from [dbo].[test_table] where [description] like @1
SET STATISTICS TIME OFF
SET STATISTICS IO OFF
 ["3000"]
```

There is third that could be used to avoid xml, and that is PROFILE I mentioned earlier, but this is returned as rowset after each batch. Here is screenshot, you need to enlarge it by clicking on image since there is a lot of columns.

![Annotation 2020-06-03 070152](https://user-images.githubusercontent.com/327155/83597664-2722df00-a568-11ea-96e4-54a51d9b6305.png)

Regarding ACTUAL execution plan, I think you can't avoid reading XML since you can't get it using only above PROFILE. So you need to use `SET STATISTICS XML ON` to get actual values (it also includes estimated)

BTW, let me know if I need to expose new function `query_multi/4` or to add flags that to existing `query` + attribute statistics in %Tds.Result{} struct

Reply:
We could easily support `json` or `rowset` formats returned as `map` or just returned as the `Result` struct of the current adapter (would be very useful for cases like MySQL traditional or TDS profile rowset), but I'm not sure about XML considering there's no `xml_library` builtin, although looks like `SET STATISTICS XML ON` is the better option for TDS because that would return more accurate statistics.


> BTW, let me know if I need to expose new function query_multi/4 or to add flags that to existing query + attribute statistics in %Tds.Result{} struct

As @josevalim comment, I believe that a variation of `query` is the way to go.